src/control-flow/function-utils.ts

```diff
@@ -0,0 +1,83 @@
+import { Query, type Node as SyntaxNode } from "web-tree-sitter";
+import type { Language } from "./cfg";
+import { extractCFunctionName } from "./cfg-c.ts";
+import { extractCppFunctionName } from "./cfg-cpp.ts";
+import { extractGoFunctionName } from "./cfg-go.ts";
+import { extractPythonFunctionName } from "./cfg-python.ts";
+import { extractTypeScriptFunctionName } from "./cfg-typescript.ts";
+
+/**
+ * Extracts the name of a node by searching for a child node with a specific type.
+ *
+ * @param func - The syntax node to search within.
+ * @param type - The type of the child node to extract the name from.
+ * used among all languages (mostly the easy cases of extracting the name).
```

Comment:
This line currently renders as part of the parameter definition, which seems odd to me.